### PR TITLE
[opentitanlib] Refactor RPC module to enable supporting future console interfaces (e.g., SPI)

### PR DESCRIPTION
**Pull request by anthonychen1251**
This PR has a commit that refactors the RPC module to support additional console devices, e.g. the SPI device console.

This PR aims to support a future OTTF SPI  device console (#21726)

### CODE REVIEW

File: sw/host/opentitanlib/src/test_utils/rpc.rs [hunk 1]
```diff
@@ -9,46 +9,59 @@ use serde::Serialize;
...
+where
+    T: ConsoleDevice + ?Sized,
+{
+    fn send(&self, device: &T) -> Result<()>;
+    fn send_with_crc(&self, device: &T) -> Result<()>;
 }
 
-impl<T: Serialize> UartSend for T {
-    fn send(&self, uart: &dyn Uart) -> Result<()> {
+impl<T, U> UartSend<T> for U
```
Comment (timothytrippel) [inline sw/host/opentitanlib/src/test_utils/rpc.rs:27]: maybe this should be `ConsoleSend` eventually? not sure how invasive changing this would be? if too much of a pain we could do it in a separate PR

File: sw/host/opentitanlib/src/test_utils/rpc.rs [hunk 2]
```diff
@@ -9,46 +9,59 @@ use serde::Serialize;
...
+    T: ConsoleDevice + ?Sized,
+{
+    fn recv(device: &T, timeout: Duration, quiet: bool) -> Result<Self>
     where
         Self: Sized;
 }
 
-impl<T: DeserializeOwned> UartRecv for T {
-    fn recv(uart: &dyn Uart, timeout: Duration, quiet: bool) -> Result<Self>
+impl<T, U> UartRecv<T> for U
```
Comment (timothytrippel) [inline sw/host/opentitanlib/src/test_utils/rpc.rs:59]: same here, `ConsoleRecv`?

**Review by timothytrippel — APPROVED** [review]

File: sw/host/opentitanlib/src/test_utils/rpc.rs (hunk 1 above)
Comment (anthonychen1251) [reply sw/host/opentitanlib/src/test_utils/rpc.rs]: You're right. I initially planned to rename this in a separate PR, but after reviewing the codebase, the changes seemed less extensive than expected. I've now included a commit for the renaming in this PR.

File: sw/host/opentitanlib/src/test_utils/rpc.rs (hunk 2 above)
Comment (anthonychen1251) [reply sw/host/opentitanlib/src/test_utils/rpc.rs]: Done.
